src/mrsimulator/spin_system/isotope.py

```
@@ -49,10 +50,86 @@ class Config:
+    def validate_symbol(cls, v, *, values, **kwargs):
+        if v in custom_isotope_data:
+            return v
         return format_isotope_string(v)
 
+    @classmethod
+    def add_new(
+        cls,
+        symbol: str,
+        spin: float,
+        gyromagnetic_ratio: float,
+        quadrupole_moment: float = 0,
+        natural_abundance: float = 100,
+        atomic_number: int = -1,
+    ):
+        """Add isotope data from a custom Isotope into the stored Isotope data and
+        return an instance of the new Isotope. The isotope symbol cannot match an real
+        isotope symbol; if the provided symbol matches a known custom isotope symbol,
+        then an instance of that isotope is returned.
+
+        Arguments:
+            (str) symbol: Required symbol for custom isotope class. String cannot match
+                another isotope symbol.
+            (float) spin: Required spin number for the isotope. Must be an integer or
+                half-integer greater than zero.
+            (float) gyromagnetic_ratio: Required gyromagnetic ratio of the isotope given
+                in MHz/T.
+            (float) quadrupole_moment: Optional quadrupole moment given in eB. Default
+                is 0.
+            (float) natural_abundance: Optional natural abundance of the isotope given
+                as a percentage between 0 and 100. Default is 100.
+            (int) atomic_number: Optional atomic number for the custom isotope. Can take
+                any integer value and has no bering on simulated spectra.
+
+        Returns:
+            An Isotope class instance
+        """
+        # Check for symbol overlap in dictionaries
+        if symbol in ISOTOPE_DATA:
+            raise ValueError(
+                "Custom isotope symbol cannot match a real isotope symbol. "
+                f"Got {symbol}."
+            )
+
+        if symbol in custom_isotope_data:
+            return Isotope(symbol=symbol)
```

If the symbol already exits in `custom_isotope_data`, it should raise an error. Consider two entries with same symbol but different spin. At present, the first entry will take priority. The better approach is to not allow duplicate symbols.

---

Remove atomic number from the custom isotopes. While real isotopes share symbol but vary in atomic number, custom isotropic should be unique symbols.

Just return `0` from `def atomic_number` when attribute is no present.